File: Algorithms/0062_Unique_Paths/Java/UniquePathsSolution1.java

// Space: O(n)
// Time: O(n)

class Solution {
    public int uniquePaths(int m, int n) {
        int[][] board = new int[m][n];
        for (int[] ints : board) {
            Arrays.fill(ints, 0);
        }
        for (int y = 0; y < m; y++) {
            for (int x = 0; x < n; x++) {
                if (y == 0 || x == 0) {
                    board[y][x] = 1;
                } else {
                    board[y][x] = board[y - 1][x] + board[y][x - 1];
                }
            }
        }
        return board[m-1][n-1];
    }
}

File: Algorithms/0062_Unique_Paths/Python/Unique_Paths_Solution_1.py

# Space: O(n)
# Time: O(n)

class Solution:
    def uniquePaths(self, m, n):
        board = [[0 for _ in range(n)] for _ in range(m)]
        for y in range(m):
            for x in range(n):
                if x == 0 or y == 0:
                    board[y][x] = 1
                else:
                    board[y][x] = board[y][x - 1] + board[y - 1][x]
        return board[-1][-1]




